include/SDL3/SDL_hints.h

```diff
@@ -2279,6 +2279,21 @@ extern "C" {
  */
 #define SDL_HINT_MAC_OPENGL_ASYNC_DISPATCH "SDL_MAC_OPENGL_ASYNC_DISPATCH"
 
+/**
+ * A variable controlling whether SDL_EVENT_MOUSE_WHEEL event values will have
+ * momentum on macOS.
+ *
+ * The variable can be set to the following values:
+ *
+ * - "0": The mouse wheel events will have no momentum. (default)
+ * - "1": The mouse wheel events will have momentum.
+ *
+ * This hint should be set before applicationDidFinishLaunching() is called.
```

Comment:
What does this mean in terms of SDL application life cycle? We should answer that for the SDL_HINT_MAC_BACKGROUND_APP hint as well. :)

Reply:
@jansedivy, can you update the documentation here?

Reply:
@slouken I did some digging to figure out when the hints are used.

There are two flows within the macOS cocoa code to call `registerUserDefaults` where the momentum is set.

1. macOS `applicationDidFinishLaunching` event that's called sometime after SDL_Init/SDL_CreateWindow. https://github.com/libsdl-org/SDL/blob/345d9bfe192493cbaf7cc547a00c0ad1d1e3bf1b/src/video/cocoa/SDL_cocoaevents.m#L333
2. Manual `registerUserDefaults` under some condition which I am not fully sure about. This is called within `SDL_Init`. https://github.com/libsdl-org/SDL/blob/345d9bfe192493cbaf7cc547a00c0ad1d1e3bf1b/src/video/cocoa/SDL_cocoaevents.m#L526-L531

This means if we want to guarantee the hint will be read in both cases it needs to be set before `SDL_Init()`. I have updated both documentation comments to reflect that. `SDL_HINT_MAC_BACKGROUND_APP` has the same code flow.

But it's still possible to set the hint after SDL_Init/SDL_CreateWindow and have `applicationDidFinishLaunching` event callback pick it up. Do we want to put some guard code to prevent that or are we good with just having the documentation say to set it before SDL_Init?